apache2/apache2_config.c

```diff
@@ -732,8 +732,10 @@ void init_directory_config(directory_config *dcfg)
     if (dcfg->col_timeout == NOT_SET) dcfg->col_timeout = 3600;
 
     /* Hash */
-    if (dcfg->crypto_key == NOT_SET_P) dcfg->crypto_key = getkey(dcfg->mp);
-    if (dcfg->crypto_key_len == NOT_SET) dcfg->crypto_key_len = strlen(dcfg->crypto_key);
+
+    /* crypto_key optimization: we only init crypto_key and crypto_key_len here. We init key when calling do_hash_link() at first time */
+    // if (dcfg->crypto_key == NOT_SET_P) dcfg->crypto_key = getkey(dcfg->mp);
```

Comment:
If this is a good and permanent change, why leave out-commented code behind? I suggest removing this fully.

Reply:
will remove it